ext/pdo_mysql/mysql_statement.c

```diff
@@ -533,15 +538,19 @@ static int pdo_mysql_stmt_param_hook(pdo_stmt_t *stmt, struct pdo_bound_param_da
 						b->buffer = &Z_DVAL_P(parameter);
 						PDO_DBG_RETURN(1);
 
-					default:
-						if (!try_convert_to_string(parameter)) {
-							PDO_DBG_RETURN(0);
+					case IS_OBJECT:
+						if(zend_class_implements_interface(Z_OBJCE_P(parameter), zend_ce_stringable)) {
+							b->buffer_type = MYSQL_TYPE_STRING;
+							b->buffer = Z_STRVAL_P(parameter);
+							b->buffer_length = Z_STRLEN_P(parameter);
+							*b->length = Z_STRLEN_P(parameter);
```

Comment:
I don't know how this works. I assume if tests pass that it works correctly.

Reply:
This is the behavior when using libmysql, so I will change the CI behavior and test it later.